third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -61,6 +61,24 @@ findEarlyInsertionPoint(Block *block, Operation *move) {
   return ipnt;
 }
 
+// Check if the operation inLoopOp is inside any scf::ForOp and outLoopOp is not
+// inside the same loop.
+bool checkLoopCondition(mlir::Operation *inLoopOp, mlir::Operation *outLoopOp) {
+  scf::ForOp parentForOp = inLoopOp->getParentOfType<scf::ForOp>();
+
+  // If the inLoopOp is not inside an scf::ForOp, return false.
```

Comment:
The following logic is effectively just `return parentForOp && !parentForOp->isAncestor(outLoopOp)`?